### PR TITLE
Update failure quality and write fail behavior

### PR DESCRIPTION
**Pull request by alexjhawk**
Update behavior of write failure to restore former value and mark quality as failure. Update quality mark to stale for realtime update failures.

### CODE REVIEW

File: eWonConnector-gateway/src/main/java/com/hms_networks/SyncManager.java [hunk 1]
```diff
@@ -818,6 +818,13 @@ else if (replaceUnderscore && !t.getName().matches(EwonConsts.ALLOWED_TAGNAME_RE
                             // Register tag write handle with provider
                             provider.registerWriteHandler(p, new WriteHandler() {
                                 public QualityCode write(TagPath tagPath, Object o) {
+                                    Object currentTagValue;
+                                    try {
+                                        currentTagValue = gatewayContext.getTagManager().readAsync(
+                                            Collections.singletonList(tagPath)).get().get(0).getValue();
```
Comment (TomKimsey) [inline eWonConnector-gateway/src/main/java/com/hms_networks/SyncManager.java]: what is the 0 in .get().get(0)?

File: eWonConnector-gateway/src/main/java/com/hms_networks/SyncManager.java [hunk 2]
```diff
@@ -818,6 +818,13 @@ else if (replaceUnderscore && !t.getName().matches(EwonConsts.ALLOWED_TAGNAME_RE
...
+                                    } catch (Exception e) {
+                                        currentTagValue = null;
```
Comment (TomKimsey) [inline eWonConnector-gateway/src/main/java/com/hms_networks/SyncManager.java]: What are the implications of setting the currentTagValue to null?  is it fine to do this?

Comment (alexjhawk) [reply eWonConnector-gateway/src/main/java/com/hms_networks/SyncManager.java]: If it's null, the value just gets set to the default of that type, so int: 0, boolean: false, etc.

File: eWonConnector-gateway/src/main/java/com/hms_networks/SyncManager.java (hunk 1 above)
Comment (alexjhawk) [reply eWonConnector-gateway/src/main/java/com/hms_networks/SyncManager.java]: magic number fixed

**Comment by TomKimsey** [comment]
With these kinds of pull requests it would probably be good to include the changelog changes as well so that its all reviewed in one place.

**Pull request merged**